src/Microsoft.TemplateEngine.Edge/BuiltInManagedProvider/GlobalSettings.cs

```diff
@@ -157,6 +163,75 @@ public async Task SetInstalledTemplatePackagesAsync(IReadOnlyList<TemplatePackag
             throw new InvalidOperationException();
         }
 
+        public async Task MigrateCacheMetadataAsync(CancellationToken cancellationToken)
+        {
+            if (!CanApplyFirstRunDataMigration())
+            {
+                return;
+            }
+
+            _logger.LogWarning(LocalizableStrings.GlobalSettingsTemplatePackageProvider_DataMigrationWarning);
+
+            string sourceFeedKey = "NuGetSource";
+            string packageIdKey = "PackageId";
+            string versionKey = "Version";
+            string ownersKey = "Owners";
+            string trustedKey = "Trusted";
+
+            var updatedPackages = new List<TemplatePackageData>();
+            var metadataDownloader = new NuGetApiPackageManager(_environmentSettings);
+            var packages = await GetInstalledTemplatePackagesAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
the idea of those method is get you the list that is modified directly if needed and then written using `SetInstalledTemplatePackagesAsync`